### PR TITLE
Implement HTML media type for NLDI resources

### PR DESCRIPTION
**Pull request by kkehl-usgs**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

Implement HTML Media Type for NLDI Resources
-----------
This is the first round for the HTML implementation, so HTML will only show a link to the JSON data.   JIRA NHGF-9.

Description
-----------
1.  User can explicitly choose HTML or JSON by using the "f" param.
2.  If the user doesn't specify, then if text/html is the first entry in the accept headers (normal case for browsers), show html.
3.  Otherwise, show JSON.

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

**Comment by kkehl-usgs** [comment]
Hmmm ... "All checks have passed" but the only thing that ran was accesslint ...

File: src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java [hunk 1]
```diff
@@ -164,4 +171,25 @@ protected String getComid(String featureSource, String featureID) {
 			return feature.get(Parameters.COMID).toString();
 		}
 	}
+
+	protected String getHtml(String url) throws IOException {
+		String html = new String(FileCopyUtils.copyToByteArray(new ClassPathResource("static/controllers/json_link_template.html").getInputStream()));
+		return html.replace(URL_MARKER, url + "?f=json");
+	}
+
+	protected String resolveFormat(String format, String acceptHeader) {
```
Comment (dsteinich) [inline src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java:180]: We should do this the Spring way - In these cases, I think using the @GetMapping's produces on two different methods (one for json, one for html). I REALLY do not think it is appropriate to produce HTML from the IOW data services. The navigation services are NOT designed for a paging interface. Not sure how they could be adapted in a reasonable generic way. The simple up/down stream main might be pagable (if a single flowline), but not the tributary navigations.

File: src/main/java/gov/usgs/owi/nldi/controllers/LinkedDataController.java [hunk 2]
```diff
@@ -82,25 +81,34 @@ public LinkedDataController(LookupDao inLookupDao, StreamingDao inStreamingDao,
...
 	}
 
-	@GetMapping(value="linked-data/{featureSource}", produces=MediaType.APPLICATION_JSON_VALUE)
-	public void getFeatures(HttpServletRequest request, HttpServletResponse response,
-							@PathVariable(LookupDao.FEATURE_SOURCE) String featureSource) {
-		BigInteger logId = logService.logRequest(request);
-
-		try {
-			Map<String, Object> parameterMap = new HashMap<>();
+	@GetMapping(value="linked-data/{featureSource}")
```
Comment (dsteinich) [inline src/main/java/gov/usgs/owi/nldi/controllers/LinkedDataController.java:84]: The Spring way would be to leave this method as it was and create a new method with produces=html and the logic specific to that.

File: src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java (hunk 1 above)
Comment (kkehl-usgs) [reply src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java:180]: Okay, I will fix the code.

Regarding the paging, as of now paging is not being discussed.  There is a github issue to create a query where the user provides latitude and longitude, and Dave Blodgett said maybe in that case we can return a leaflet and the user can see the points in the basin where that latitude and longitude is contained.  What to do for the navigation stuff is still very TBD.   But I will keep your words in mind.

Comment (kkehl-usgs) [reply src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java]: Code is fixed.

**Comment by dblodgett-usgs** [comment]
@dsteinich @kkehl-usgs -- as noted in #135, let's defer the the HTML media type till we've had more time to discuss the public API media type strategy. Is there anything in this PR that should be brought forward whether we pursue that or not?

**Comment by kkehl-usgs** [comment]
This work has been deferred.

**Pull request closed without merging**